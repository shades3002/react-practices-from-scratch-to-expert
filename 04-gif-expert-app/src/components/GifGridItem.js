import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ id, title, url}) => {

    return (
        <div className="card animate__animated animate__bounce">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}   