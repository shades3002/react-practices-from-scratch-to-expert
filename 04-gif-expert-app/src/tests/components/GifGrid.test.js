import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');


describe('Test <GifGrid />', () => {

    const category = '';
    
    test('should show correctly', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('should show item when images are loaded from useFetchGifs', () => {
        const gifs = [{
            id: 'Ans',
            url: 'https://localhost/ans.jpg',
            title: 'ans'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'p' ).exists() ).toBe( false );
        expect( wrapper.find( 'GifGridItem' ).length ).toBe( gifs.length );
        
    });
        
});
