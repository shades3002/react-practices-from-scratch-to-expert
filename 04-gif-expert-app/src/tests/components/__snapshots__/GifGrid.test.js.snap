// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> should show correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__backInDown"
  />
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid animate__animated animate__fadeInUp"
  />
</Fragment>
`;

exports[`Test <GifGrid /> should show item when images are loaded from useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__backInDown"
  />
  <div
    className="card-grid animate__animated animate__fadeInUp"
  >
    <GifGridItem
      id="Ans"
      key="Ans"
      title="ans"
      url="https://localhost/ans.jpg"
    />
  </div>
</Fragment>
`;
